openquake/hazardlib/gsim/kuehn_2020.py

```diff
@@ -0,0 +1,1260 @@
+    CONSTS = {"c_10": 0.0,
+              "c": 1.88,
+              "n": 1.18,
+              "delta_m": 0.1,
+              "delta_z": 1,
+              "Mref": 6.0,
+              "Zref": 15,
+              "z_b": 30,
+              "m_b": 8.6}
+
+class KuehnEtAl2020SInterCascadia(KuehnEtAl2020SInter):
+    """
+    Implements the regional model for Cascadia
+    """
+    #: Required site parameters are Vs30, Vs30 type (measured or inferred),
+    #: and depth (km) to the 2.5 km/s shear wave velocity layer (z2pt5)
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def get_mean_values(self, C, m_b, sites, rup, dists, a1100):
+        """
+        Returns the mean values for a specific IMT
+        """
+        if isinstance(a1100, np.ndarray):
+            # Site model defined
+            temp_vs30 = sites.vs30
+            temp_z = sites.z2pt5
+        else:
+            # Default site and basin model
+            temp_vs30 = 1100.0 * np.ones(len(sites.vs30))
+            temp_z = -1 * np.ones(len(sites.vs30))
+
+        return (self.get_base_term(C) +
+                self.get_magnitude_scaling_term(C, m_b, rup.mag) +
+                self.get_geometric_attenuation_term(C, rup.mag, dists.rrup) +
+                self.get_shallow_site_response_term(C, temp_vs30, a1100) +
+                self.get_basin_response_term(C, temp_z, temp_vs30) +
+                self.get_depth_term(C, rup.ztor) +
+                self.get_anelastic_attenuation_term(C, dists))
+
+    def get_base_term(self, C):
+        """
+        Returns the constant coefficient for Cascadia
+        """
+        return C["c_1_if_reg_Ca"]
+
+    def _get_anelastic_coeff(self, C):
+        """
+        Returns the anelastic coefficient for Cascadia
+        """
+        return C["c_6_2_reg_Ca"]
+
+    def _get_linear_site_coeff(self, C):
+        """
+        Returns the linear site scaling coefficient for Cascadia
+        """
+        return C["c_7_reg_Ca"]
+
+    def get_basin_response_term(self, C, z_value, vs30):
+        """
+        Returns the basin response term (Eq. 4.10)
+        Internal scale of Z2.5 is in meters
+        """
+        mask = (z_value > 0)
+        brt = np.zeros_like(z_value)
+        brt[mask] = C["c_11_Ca"] + C["c_12_Ca"] * \
+                    (np.log(z_value[mask] * 1000.) - self._get_ln_z_ref(vs30[mask]))
+        return brt
+
+    CONSTS = {"c_10": 0.0,
+              "c": 1.88,
+              "n": 1.18,
+              "delta_m": 0.1,
+              "delta_z": 1,
+              "Mref": 6.0,
+              "Zref": 15,
+              "z_b": 30,
+              "m_b": 8.}
+
+    CONSTS_Z_MODEL = {"c_z_1": 8.294049640102028,
+                      "c_z_2": 2.302585092994046,
+                      "c_z_3": 6.396929655216146,
+                      "c_z_4": 0.27081458999999997}
+
+class KuehnEtAl2020SInterCentralAmericaMexico(KuehnEtAl2020SInter):
+    """
+    Implements the regional interface model for Central America and Mexico
+    """
+
```

Comment:
oq has a feature where if you add

`experimental = True` 

into a class, then the engine raises a warning if anyone tries to use it in hazard calculations. Could be good to add this into the relevant regional classes, so that we don't forget to address this later

Reply:
OK, I'll add experimental to all classes, because it's not finished yet. I'll get to it tomorrow.